op-e2e/actions/proofs/operator_fee_test.go

```diff
@@ -54,55 +111,110 @@ func Test_Operator_Fee_Constistency(gt *testing.T) {
+			env.Engine.ActL2IncludeTx(env.Alice.Address())(t)
+			env.Sequencer.ActL2EndBlock(t)
+			env.Alice.L2.ActCheckReceiptStatusOfLastTx(true)(t)
+
+			aliceInitialBalance, l1FeeVaultInitialBalance, baseFeeVaultInitialBalance, sequencerFeeVaultInitialBalance, operatorFeeVaultInitialBalance = getCurrentBalances()
+
+			// regular Deposit, in new L1 block
+			env.Alice.L1.ActResetTxOpts(t)
+			env.Alice.ActDeposit(t)
+			env.Miner.ActL1StartBlock(12)(t)
+			env.Miner.ActL1IncludeTx(env.Alice.Address())(t)
+			env.Miner.ActL1EndBlock(t)
+
+			// sync sequencer build enough blocks to adopt latest L1 origin
+			env.Sequencer.ActL1HeadSignal(t)
+			env.Sequencer.ActBuildToL1HeadUnsafe(t)
+
+			receipt, err = env.Alice.GetLastDepositL2Receipt(t)
+			require.NoError(t, err)
+		}
 
-		l1FeeVaultBalance := balanceAt(predeploys.L1FeeVaultAddr)
-		baseFeeVaultBalance := balanceAt(predeploys.BaseFeeVaultAddr)
-		sequencerFeeVaultBalance := balanceAt(predeploys.SequencerFeeVaultAddr)
-		operatorFeeVaultFinalBalance := balanceAt(predeploys.OperatorFeeVaultAddr)
-		aliceFinalBalance := balanceAt(env.Alice.Address())
+		aliceFinalBalance, l1FeeVaultFinalBalance, baseFeeVaultFinalBalance, sequencerFeeVaultFinalBalance, operatorFeeVaultFinalBalance := getCurrentBalances()
 
-		require.True(t, aliceFinalBalance.Cmp(aliceInitialBalance) < 0, "Alice's balance should decrease")
+		if receipt == nil {
+			receipt = env.Alice.L2.LastTxReceipt(t)
+		}
 
-		// Check that the operator fee sent to the vault is correct
-		require.Equal(t,
-			new(big.Int).Add(
-				new(big.Int).Div(
-					new(big.Int).Mul(
-						new(big.Int).SetUint64(receipt.GasUsed),
-						new(big.Int).SetUint64(uint64(testOperatorFeeScalar)),
+		if testCfg.Custom == DepositTx || testCfg.Custom == IsthmusTransitionBlock {
+			require.Nil(t, receipt.OperatorFeeScalar)
+			require.Nil(t, receipt.OperatorFeeConstant)
+
+			// Nothing should has been sent to operator fee vault
+			require.Equal(t, operatorFeeVaultInitialBalance, operatorFeeVaultFinalBalance)
+
+		} else {
+			// Check that the operator fee was applied
+			require.Equal(t, testOperatorFeeScalar, uint32(*receipt.OperatorFeeScalar))
+			require.Equal(t, testOperatorFeeConstant, *receipt.OperatorFeeConstant)
+
+			// Check that the operator fee sent to the vault is correct
+			require.Equal(t,
+				new(big.Int).Add(
+					new(big.Int).Div(
+						new(big.Int).Mul(
+							new(big.Int).SetUint64(receipt.GasUsed),
+							new(big.Int).SetUint64(uint64(testOperatorFeeScalar)),
+						),
+						new(big.Int).SetUint64(1e6),
 					),
-					new(big.Int).SetUint64(1e6),
+					new(big.Int).SetUint64(testOperatorFeeConstant),
 				),
-				new(big.Int).SetUint64(testOperatorFeeConstant),
-			),
-			operatorFeeVaultFinalBalance,
-		)
+				new(big.Int).Sub(operatorFeeVaultFinalBalance, operatorFeeVaultInitialBalance),
+			)
+
+			require.True(t, aliceFinalBalance.Cmp(aliceInitialBalance) < 0, "Alice's balance should decrease")
```

Comment:
can we make an assertion about alice's balance in the deposit case? what does the deposit cause?

Reply:
I've checked `aliceInitialBalance` and `aliceFinalBalance` in the deposit case and they are equals, I don't understand why

Reply:
This would be good to figure out before merge

Reply:
Fixed

Reply:
I moved the check for alice's balance out of the if/else branch so it runs for all cases. And that's successful.